### PR TITLE
feat(FR-1714): implement AI-powered Playwright test automation workflow

### LINKED ISSUE #4683: Implement AI-Powered Playwright Test Automation Workflow with Chat Mode Agents
## 🎯 **Goal**

Implement AI-powered Playwright test automation workflow using specialized chat mode agents.

### **Automated Test Creation & Maintenance**

- **Plan Tests**: AI agent analyzes web UI and creates comprehensive test scenarios
- **Generate Code**: AI agent converts test plans into working Playwright test code
- **Fix Broken Tests**: AI agent debugs and repairs failing tests automatically

### **Expected Benefits**

- **5x faster** test creation compared to manual writing
- **60% reduction** in test maintenance time
- **Consistent quality** across all E2E tests
- **Lower barrier to entry** for non-technical team members

## 🔄 **Simple Workflow**

```
1. Plan → AI creates test scenarios by exploring UI
2. Generate → AI writes Playwright code from scenarios  
3. Heal → AI fixes broken tests when code changes
```

## 📊 **Success Criteria**

- Developers can create comprehensive E2E tests without manual coding
- Test failures are automatically diagnosed and fixed
- New features get full test coverage within minutes, not hours

This will significantly improve our testing efficiency and code quality! 🚀

JIRA Issue: FR-1714

### PR DESCRIPTION
**Pull request by yomybaby**
resolves #4683 (FR-1714)

## Summary

This PR implements an AI-powered Playwright test automation workflow using three specialized chat mode agents that automate the entire test lifecycle - from planning to generation to healing.

### Goal
- **5x faster test creation** through automated test scenario generation
- **60% reduction in maintenance time** through automated test healing
- **Improved test coverage** through comprehensive test planning

### Three Specialized Agents

1. **🎭 Planner Agent** (`.claude/agents/playwright-test-planner.md`)
   - Analyzes web UI through browser snapshots
   - Creates comprehensive test scenarios covering happy paths, edge cases, and error conditions
   - Outputs structured test plans in markdown format

2. **🎭 Generator Agent** (`.claude/agents/playwright-test-generator.md`)
   - Converts test plans into working Playwright code
   - Follows best practices and existing test patterns
   - Uses utility classes and helper functions from the codebase

3. **🎭 Healer Agent** (`.claude/agents/playwright-test-healer.md`)
   - Debugs and repairs failing tests automatically
   - Analyzes error messages, screenshots, and traces
   - Applies fixes and verifies test passes

### Practical Demonstration: VFolder E2E Tests

This PR includes a complete demonstration of the workflow with VFolder-related tests:

**New Utility Classes:**
- `FolderExplorerModal` - Page object class for folder explorer modal interactions
- `NotificationHandler` - Centralized notification management to prevent UI blocking

**New Test Specifications:**
- `vfolder-explorer-modal.spec.ts` - Comprehensive tests for folder explorer modal (6 tests)
- `vfolder-consecutive-deletion.spec.ts` - Tests for consecutive folder operations

**Enhanced Test Utilities:**
- Improved `deleteForeverAndVerifyFromTrash()` with filter clearing and better error handling
- Added `clearAllFilters()` helper function

**Configuration:**
- Added Playwright MCP server to `.mcp.json` for agent integration
- GitHub Copilot chat modes for easy agent access

### Technical Implementation

**Agent Architecture:**
- Each agent has specialized tools and focused responsibilities
- Agents use Playwright MCP server for browser automation
- Planner → Generator → Healer workflow for complete test lifecycle

**Benefits for Developers:**
1. **Fast Test Creation**: Generate comprehensive test plans and code from web UI exploration
2. **Reduced Maintenance**: Automated healing fixes broken tests caused by UI changes
3. **Better Coverage**: AI-powered planning identifies edge cases humans might miss
4. **Consistent Patterns**: Generated code follows established patterns and utilities

**Checklist:**

- [x] Documentation - Agent markdown files with detailed instructions
- [ ] Minimum required manager version - N/A (E2E test infrastructure only)
- [ ] Specific setting for review - Requires Playwright and MCP server setup
- [x] Test case(s) to demonstrate - VFolder tests demonstrate the complete workflow
- [x] Minimum requirements to check during review:
  - Agent files in `.claude/agents/` are comprehensive and follow best practices
  - Utility classes (FolderExplorerModal, NotificationHandler) are well-structured
  - Test specs follow existing patterns and pass consistently
  - MCP server configuration is correct

### CODE REVIEW

File: e2e/utils/classes/FolderExplorerModal.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+  async waitForOpen(): Promise<void> {
+    await expect(this.modal).toBeVisible();
+  }
+
+  async clickCloseButton(): Promise<void> {
+    const closeButton = this.modal.getByRole('button', { name: 'Close' });
+    await expect(closeButton).toBeVisible();
+    await closeButton.click();
+  }
```
Comment (nowgnuesLee) [inline e2e/utils/classes/FolderExplorerModal.ts:19]: Let’s use getCloseButton instead of defining it manually

File: e2e/vfolder-consecutive-deletion.spec.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    // Create three folders
+    // Dismiss notifications after each creation to prevent UI blocking
+    await createVFolderAndVerify(page, folder1Name);
+    await notification.closeAllNotifications();
+
+    await createVFolderAndVerify(page, folder2Name);
+    await notification.closeAllNotifications();
+
+    await createVFolderAndVerify(page, folder3Name);
+    await notification.closeAllNotifications();
```
Comment (nowgnuesLee) [inline e2e/vfolder-consecutive-deletion.spec.ts:35]: Could you refactor createVFolderAndVerify to use FolderCreationModal class?

File: e2e/vfolder-explorer-modal.spec.ts [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+    // Only create the folder once for all tests in this serial group
+    if (!folderCreated) {
+      await createVFolderAndVerify(page, testFolderName);
+      folderCreated = true;
+    } else {
+      // Just navigate to the data page if folder already exists
+      await navigateTo(page, 'data');
+    }
+  });
+
```
Comment (nowgnuesLee) [inline e2e/vfolder-explorer-modal.spec.ts]: how about using beforeAll to create a folder and move to data page instead of using folderCreated var. And I think the part that opens the modal should be in beforeEach

**Review by ironAiken2 — CHANGES_REQUESTED** [review]
It would be good to add a convention for test names. Previously, test names were written from the user's perspective to match e2e tests, but now they are written as if they were unit tests. Also, since tests generated by AI are written in separate files, it would be better if they were written under existing tests.  

Everything else works fine :)

File: e2e/vfolder-explorer-modal.spec.ts [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+});
+
+test.describe.serial('FolderExplorerModal - User Modal Interaction', () => {
+  const testFolderName = 'e2e-test-folder-layout-' + new Date().getTime();
+
+  test.beforeAll(async ({ page }) => {
+    // Create the folder once before all tests in this serial group
+    await loginAsUser(page);
+    await createVFolderAndVerify(page, testFolderName);
+  });
```
Comment (nowgnuesLee) [inline e2e/vfolder-explorer-modal.spec.ts]: we can't use page fixture in `beforeAll`​.

```suggestion
  test.beforeAll(async ({ page }) => {
    // Create the folder once before all tests in this serial group
    const context = await browser.newContext();
    const page = await context.newPage();
    await loginAsUser(page);
    await createVFolderAndVerify(page, testFolderName);
  });
```

File: e2e/vfolder-explorer-modal.spec.ts [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+    await modal.verifyFileExplorerNotLoaded();
+
+    // 4. Verify modal can still be closed
+    const closeButton = await modal.getCloseButton();
+    // For React development mode, the button may be covered by Red screen; use force click
+    await closeButton.click({ force: true });
+  });
+});
+
+test.describe.serial('FolderExplorerModal - User Modal Interaction', () => {
```
Comment (nowgnuesLee) [inline e2e/vfolder-explorer-modal.spec.ts]: **Filebrowser tests can fail depending on the test environment, so why are they run serially?**

Comment (yomybaby) [reply e2e/vfolder-explorer-modal.spec.ts]: No reason to use `serial`​.

File: e2e/vfolder-explorer-modal.spec.ts (hunk 4 above)
Comment (yomybaby) [reply e2e/vfolder-explorer-modal.spec.ts]: I changed and submited. Please check again. Thank you.

File: e2e/vfolder-explorer-modal.spec.ts [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+test.describe('FolderExplorerModal - User Modal Interaction', () => {
+  const testFolderName = 'e2e-test-folder-layout-' + new Date().getTime();
+
+  test.beforeAll(async ({ browser }) => {
+    // Create the folder once before all tests in this serial group
+    const context = await browser.newContext();
+    const page = await context.newPage();
+    await loginAsUser(page);
+    await createVFolderAndVerify(page, testFolderName);
+  });
```
Comment (nowgnuesLee) [inline e2e/vfolder-explorer-modal.spec.ts:150]:
```suggestion
  test.beforeAll(async ({ browser }) => {
    // Create the folder once before all tests in this serial group
    const context = await browser.newContext();
    const page = await context.newPage();
    await loginAsUser(page);
    await createVFolderAndVerify(page, testFolderName);
    await context.close();
  });
```

**Review by nowgnuesLee — CHANGES_REQUESTED** [review]

**Review by nowgnuesLee — APPROVED** [review]
LGTM
